src/expr/src/interpret.rs

```diff
@@ -787,29 +787,16 @@ impl<'a> Interpreter for ColumnSpecs<'a> {
     }
 
     fn variadic(&self, func: &VariadicFunc, args: Vec<Self::Summary>) -> Self::Summary {
-        let mapped_spec = if func.is_associative() && args.len() >= 2 {
-            let col_type = func.output_type(args.iter().map(|cs| cs.col_type.clone()).collect());
-            let mut expr = MirScalarExpr::CallVariadic {
-                func: func.clone(),
-                exprs: vec![
-                    Self::placeholder(col_type.clone()),
-                    Self::placeholder(col_type),
-                ],
-            };
-            let is_monotone = func.is_monotone();
-            args.iter()
-                .map(|cs| cs.range.clone())
-                .reduce(|left, right| {
-                    left.flat_map(is_monotone, |left_result| {
-                        Self::set_argument(&mut expr, 0, left_result);
```

Comment:
On the second call to this reduce function, we'll end up trying to set the first argument to `null` -- since the `ResultSpec` returned by the first call included null -- and trigger our safety check.